Improve cli --help

Nodraak opened the pull request:
Hi!

*First of all, thanks a lot for this tool!*

I think the cli help could be improved. Here is an attempt to fix:
* The old default format is not indicated. Changing the default format broke my wrapper and it took me a bit of investigation to find out.
* Daca2 format is not indicated in the help but available in the source code

danmar commented on cli/cmdlineparser.cpp:
```diff
@@ -1161,8 +1161,7 @@ void CmdLineParser::printHelp()
               "                         '{callstack} {message}'\n"
-              "                         Pre-defined templates: gcc, vs, edit, cppcheck1\n"
-              "                         The default format is 'gcc'.\n"
+              "                         Pre-defined templates: gcc (default), cppcheck1 (old default), vs, edit, daca2.\n"
```
I would prefer that `daca2` is not shown even though it is not "secret". It was not designed to be used by users. If we want to change it for our scripts it would feel good if we can do that without any concern for users.

Nodraak replied:
Maybe we can indicate that it is an internal api or something?

danmar replied:
I'm not sure.. what is the point to show it in the help if it's for internal use.

Nodraak commented:
Hi again, sorry for the late response.

I updated my PR with your preference. I added a comment mentionning daca2, so it is not visible to users, but dev might find it and therefore understand why it is not shown in the `--help`.

I hope this compromise is satisfactory

danmar commented:
yes. sure. it is not "secret" just "not official". Thanks for tweaking our help!
